test/langtools/tools/javac/patterns/NestedPatternVariablesBytecode.java

```diff
@@ -0,0 +1,114 @@
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8268748
+ * @summary Javac generates error opcodes when using nest pattern variables
+ * @library /tools/lib
+ * @modules jdk.compiler/com.sun.tools.javac.api
+ *          jdk.compiler/com.sun.tools.javac.main
+ *          jdk.jdeps/com.sun.tools.classfile
+ * @build toolbox.ToolBox toolbox.JavacTask
+ * @run main NestedPatternVariablesBytecode
+ */
+
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.StreamSupport;
+
+import com.sun.tools.classfile.ClassFile;
+import com.sun.tools.classfile.ConstantPoolException;
+import com.sun.tools.classfile.Method;
+import com.sun.tools.classfile.Attribute;
+import com.sun.tools.classfile.Code_attribute;
+import com.sun.tools.classfile.Instruction;
+
+import toolbox.JavacTask;
+import toolbox.TestRunner;
+import toolbox.ToolBox;
+
+public class NestedPatternVariablesBytecode extends TestRunner {
+    private static final String JAVA_VERSION = System.getProperty("java.specification.version");
+    private static final String TEST_METHOD = "test";
+
+    ToolBox tb;
+    ClassFile cf;
+
+    public NestedPatternVariablesBytecode() {
+        super(System.err);
+        tb = new ToolBox();
+    }
+
+    public static void main(String[] args) throws Exception {
+        NestedPatternVariablesBytecode t = new NestedPatternVariablesBytecode();
+        t.runTests();
+    }
+
+    @Test
+    public void testNestedPatternVariablesBytecode() throws Exception {
+        String code = """
+                class NestedPatterVariablesTest {
+                    String test(Object o) {
+                        if (o instanceof (CharSequence cs && cs instanceof String s)) {
+                            return s;
+                        }
+                        return null;
+                    }
+                }""";
+        Path curPath = Path.of(".");
+        new JavacTask(tb)
+                .options("--enable-preview", "-source", JAVA_VERSION)
+                .sources(code)
+                .outdir(curPath)
+                .run();
+
+        cf = ClassFile.read(curPath.resolve("NestedPatterVariablesTest.class"));
+        Method testMethod = Arrays.stream(cf.methods)
+                                  .filter(m -> isTestMethod(m))
+                                  .findAny()
+                                  .get();
+        Code_attribute code_attribute = (Code_attribute) testMethod.attributes.get(Attribute.Code);
+
+        List<String> actualCode = getCodeInstructions(code_attribute);
+        List<String> expectedCode = Arrays.asList(
+                "aload_1", "instanceof", "ifeq", "aload_1", "checkcast", "astore_2", "aload_2", "instanceof",
```

Comment:
not sure I like this code pattern check, if we change the generated code for whatever reason we will need to revisit this test. Is there other way of testing the same? like executing a code and checking its results?

Reply:
@vicente-romero-oracle Thanks for your review.

In this test, if the bug is not fixed which means the `aload_2` would be `aload_1`, the code runs well too.
The `aload_1` or `aload_2` don't effect the result so that we can't check it by executing it.
There are some similar situations in the past when we want to verify the class files.
These tests would also fail if we change the logic of the code generation.

Another way is that only verify if the `aload_2` is at the right place instead of verify all the instruments.
But it also need to be revisited and revised if the logic of the code generation is changed in the future.
It seems a unavoidable issue about verifing the class files.

Reply:
> @vicente-romero-oracle Thanks for your review.
> 
> In this test, if the bug is not fixed which means the `aload_2` would be `aload_1`, the code runs well too.
> The `aload_1` or `aload_2` don't effect the result so that we can't check it by executing it.
> There are some similar situations in the past when we want to verify the class files.
> These tests would also fail if we change the logic of the code generation.
> 
> Another way is that only verify if the `aload_2` is at the right place instead of verify all the instruments.
> But it also need to be revisited and revised if the logic of the code generation is changed in the future.
> It seems a unavoidable issue about verifing the class files.

I see, thanks for the clarification